tests/test_abstract_repr.py

```diff
@@ -264,10 +287,32 @@ def test_noise_model(noise_model: NoiseModel):
             ser_noise_model_obj[param] = (
                 ser_noise_model_obj[param] or _LEGACY_DEFAULTS[param]
             )
-    assert (
-        NoiseModel.from_abstract_repr(json.dumps(ser_noise_model_obj))
-        == re_noise_model
+    legacy_noise_model = NoiseModel.from_abstract_repr(
+        json.dumps(ser_noise_model_obj)
     )
+    dict_legacy = asdict(legacy_noise_model)
+    dict_re_noise_model = asdict(re_noise_model)
+    # The noise models are equivalent: doppler isn't among noise types if
+    # temperature is now >0 whereas it was 0 in initial noise model.
+    disable_doppler = dict_legacy.pop("disable_doppler")
+    assert disable_doppler == (
+        dict_legacy["temperature"] > 0
+        and "doppler" not in re_noise_model.noise_types
+    )
+    dict_re_noise_model.pop("disable_doppler")
+    if disable_doppler:
+        # Temperature/runs have been updated if they were 0/None
+        temp = dict_legacy.pop("temperature")
+        assert temp == (
+            noise_model.temperature or _LEGACY_DEFAULTS["temperature"]
+        )
+        dict_re_noise_model.pop("temperature")
+        assert (
+            dict_legacy.pop("runs") == noise_model.runs
+            or _LEGACY_DEFAULTS["runs"]
+        )
+        dict_re_noise_model.pop("runs")
```

Comment:
The purpose of this part of the test was to make sure old noise models are still deserialized correctly, so allowing `disable_noise_model` to be `True` in the legacy does not make much sense imo... 
Instead of this complicated thing, I'd consider splitting this "legacy" section into a separate test where you purposefully specify only the parameters and noise types that were there before v0.20 (ie those in `_LEGACY_DEFAULTS`)

Reply:
Damn, ok... :disappointed: